matchmaker/views/matchmaker_api_tests.py

```diff
@@ -367,40 +347,45 @@ def test_search_individual_mme_matches(self, mock_nodes, mock_email, mock_logger
-            'mmeResultGuids': mock.ANY,
-            'individualGuid': INDIVIDUAL_GUID,
-            'submissionGuid': SUBMISSION_GUID,
-            'createdDate': '2018-05-23T09:07:49.719Z',
-            'lastModifiedDate': '2018-05-23T09:07:49.719Z',
-            'deletedDate': None,
-            'contactName': 'Sam Baxter',
-            'contactHref': 'mailto:matchmaker@broadinstitute.org,test_user@broadinstitute.org',
-            'submissionId': 'NA19675_1_01',
-            'phenotypes': [
-                {'id': 'HP:0001252', 'label': 'Muscular hypotonia', 'observed': 'yes'},
-                {'id': 'HP:0001263', 'label': 'Global developmental delay', 'observed': 'no'},
-                {'id': 'HP:0012469', 'label': 'Infantile spasms', 'observed': 'yes'}
-            ],
-            'geneVariants': [{
-                'geneId': 'ENSG00000135953',
-                'variantGuid': 'SV0000001_2103343353_r0390_100',
-            }],
-        }})
-        self.assertDictEqual(response_json['individualsByGuid'], {INDIVIDUAL_GUID: {
-            'mmeSubmissionGuid': SUBMISSION_GUID,
-        }})
-        self.assertSetEqual(
-            set(response_json['mmeSubmissionsByGuid'][SUBMISSION_GUID]['mmeResultGuids']),
-            set(response_json['mmeResultsByGuid'].keys()))
 
         self.assertSetEqual(
             set(response_json['genesById'].keys()),
             {'ENSG00000135953', 'ENSG00000240361', 'ENSG00000223972'}
         )
+        self.assertDictEqual(response_json['mmeContactNotes'], {})
+
+        # Test external matches
+        node_a_match_url = reverse(search_individual_mme_matches, args=[SUBMISSION_GUID, 'Node A'])
+        response = self.client.get(node_a_match_url, {'incomingQueryGuid': incoming_query_guid})
+        self.assertEqual(response.status_code, 400)
+        self.assertListEqual(response.json()['errors'],  ['Error searching in Node A: Failed request (400)'])
+
+        node_b_match_url = reverse(search_individual_mme_matches, args=[SUBMISSION_GUID, 'Node B'])
+        response = self.client.get(node_b_match_url, {'incomingQueryGuid': incoming_query_guid})
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertSetEqual(set(response_json.keys()), {'mmeResultsByGuid', 'genesById', 'mmeContactNotes'})
+
+        self.assertEqual(len(response_json['mmeResultsByGuid']), 1)
+        new_result_guid = next(k for k in response_json['mmeResultsByGuid'].keys())
+        self.assertDictEqual(response_json['mmeResultsByGuid'][new_result_guid], PARSED_NEW_MATCH_JSON)
+        self.assertSetEqual(set(response_json['genesById'].keys()), {'ENSG00000135953'})
         # non-analyst users can't see contact notes
         self.assertDictEqual(response_json['mmeContactNotes'], {'st georges, university of london': {}})
 
+        # Test notifications and removed result cleanup
+        finalize_search_url = reverse(finalize_mme_search, args=[SUBMISSION_GUID])
```

Comment:
The `finalize_mme_search` API may need a separate test function to test the permission and async calls with the `search_individual_mme_matches` API.

Reply:
But this endpoint simply won't work if its called outside the context of a search, and thats fine, since that can never happen in the UI and as such if someone does manage to call it incorrectly by like inspecting the network pnale and copying the URL we don't care what response they get back as they should not have done that. We want to test error cases where we actually care about the result and want to maintain the error handling behavior going forward, but for cases we don't care about theres no point in writing unit tests to enforce arbitrary behavior.